app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -83,14 +85,24 @@ class FormController(
                    cache.form.thirdPartyData,
                    cache.formTemplate)
                    .map {
-                     case (validationResult, v, _) =>
-                       (
-                         ValidationUtil.isFormValid(validationResult.toMap),
-                         FormData(validationResult.flatMap(_._2.toFormField)),
-                         v)
+                     case (validationResult, validatedType, _) =>
+                       validateFormHelper(validationResult, validatedType)
+
                    }
     } yield formData
 
+  private def validateFormHelper(
+    validationResult: List[(FormComponent, FormFieldValidationResult)],
+    validatedType: ValidatedType[ValidationResult]): FormValidationOutcome = {
+    val isFormValid = ValidationUtil.isFormValid(validationResult.toMap)
+    val formComponents =
+      if (isFormValid) formService.removeCommas(validationResult) else validationResult
```

Comment:
We cannot remove commas here, since this affect calculated values and data entered by user as well. Commas needs to be removed before we generate handlebar payload.